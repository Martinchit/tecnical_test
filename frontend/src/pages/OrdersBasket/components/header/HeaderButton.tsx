import styled from 'styled-components';
import { Button } from 'react-bootstrap';

export const HeaderButton = styled(Button)`
  margin-left: 2%;
  width: 10%;

  @media (max-width: 767px) {
    width: 45%;
    margin-top: 3%;
    font-size: 0.8rem;
  }

  &:focus {
    outline: none;
    box-shadow: none;
  }
`;
