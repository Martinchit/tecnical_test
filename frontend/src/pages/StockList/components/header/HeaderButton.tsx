import styled from 'styled-components';
import { Button } from 'react-bootstrap';

export const HeaderButton = styled(Button)`
  margin-left: 1%;
  width: 10%;
  min-width: 85px;

  @media (max-width: 767px) {
    width: 50%;
    margin-top: 5%;
    font-size: 0.8rem;
  }

  &:focus {
    outline: none;
    box-shadow: none;
  }
`;
