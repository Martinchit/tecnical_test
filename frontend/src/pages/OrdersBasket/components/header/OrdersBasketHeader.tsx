import React from 'react';
import { HeaderContainer } from './HeaderContainer';
import { HeaderButton } from './HeaderButton';

interface OrdersBasketHeaderProps {
  onRemoveButtonClickHandler: () => void;
  onBookButtonClickHandler: () => void;
  removeButtonDisable: boolean;
  bookButtonDisable: boolean;
}

export const OrdersBasketHeader: React.FC<OrdersBasketHeaderProps> = ({
  onRemoveButtonClickHandler,
  onBookButtonClickHandler,
  removeButtonDisable,
  bookButtonDisable,
}) => {
  return (
    <HeaderContainer>
      <HeaderButton
        disabled={removeButtonDisable}
        variant="primary"
        onClick={onRemoveButtonClickHandler}
      >
        Remove
      </HeaderButton>
      <HeaderButton
        disabled={bookButtonDisable}
        variant="primary"
        onClick={onBookButtonClickHandler}
      >
        Book
      </HeaderButton>
    </HeaderContainer>
  );
};
