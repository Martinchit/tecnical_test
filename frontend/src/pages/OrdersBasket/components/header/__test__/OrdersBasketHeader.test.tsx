import React from 'react';
import { shallow } from 'enzyme';

import { OrdersBasketHeader } from '../OrdersBasketHeader';
import { HeaderButton } from '../HeaderButton';

describe('HeaderContainer', () => {
  const props = {
    onRemoveButtonClickHandler: jest.fn(),
    onBookButtonClickHandler: jest.fn(),
    removeButtonDisable: true,
    bookButtonDisable: true,
  };

  const restoreMock = () => {
    props.onRemoveButtonClickHandler.mockRestore();
    props.onBookButtonClickHandler.mockRestore();
  };

  it('renders correctly', () => {
    const wrapper = shallow(<OrdersBasketHeader {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('triggers onRemoveButtonClickHandler', () => {
    const wrapper = shallow(<OrdersBasketHeader {...props} />);
    wrapper.find(HeaderButton).first().simulate('click');
    expect(props.onRemoveButtonClickHandler).toHaveBeenCalled();
    restoreMock();
  });

  it('triggers onBookButtonClickHandler', () => {
    const wrapper = shallow(<OrdersBasketHeader {...props} />);
    wrapper.find(HeaderButton).at(1).simulate('click');
    expect(props.onBookButtonClickHandler).toHaveBeenCalled();
    restoreMock();
  });
});
