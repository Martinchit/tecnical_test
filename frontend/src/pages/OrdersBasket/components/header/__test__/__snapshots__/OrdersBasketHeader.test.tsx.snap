// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderContainer renders correctly 1`] = `
<Styled(styled.div)>
  <Styled(Button)
    disabled={true}
    onClick={[MockFunction]}
    variant="primary"
  >
    Remove
  </Styled(Button)>
  <Styled(Button)
    disabled={true}
    onClick={[MockFunction]}
    variant="primary"
  >
    Book
  </Styled(Button)>
</Styled(styled.div)>
`;
